lib/engines/hcktest/tests.rb

```diff
@@ -132,11 +132,22 @@ def wait_queued_test(id)
       end
     end
 
-    def test_parameters(test_name)
+    def read_test_config(test_name, config)
       @project.engine.config.tests_config
               .select { _1.tests.include?(test_name) }
-              .flat_map(&:parameters)
-              .to_h { |parameter| [parameter.name, parameter.value] }
+              .flat_map(&config)
+    end
+
+    def test_parameters(test_name)
+      read_test_config(test_name, :parameters)
+        .to_h { |parameter| [parameter.name, parameter.value] }
+    end
+
+    def run_test_commands(test_name, type)
+      read_test_config(test_name, type).each do |commands|
+        @tools.run_on_machine(@client.name, commands.guest, commands.guest) if commands.guest
+        spawn(*commands.host, out: :err) if commands.host
```

Comment:
Do these commands run synchronously or asynchronously?
Why do not use the CmdRun class?